### PR TITLE
risc-v/esp32c3: Support ESP32-C3 efuse

### PR DESCRIPTION
**Pull request by lhespress**
## Summary
risc-v/esp32c3: Support ESP32-C3 efuse Driver
## Testing
The efuse driver was successfully tested in esp32c3-devkitc using the efuse example.
- config

  ./tools/configure esp32c3-devkit:efuse

- test result

  efuse

### CODE REVIEW

**Comment by gustavonihei** [comment]
@lhespress Please, use the PR description template.

**Comment by lhespress** [comment]
> @lhespress Please, use the PR description template.

Modify it.

**Comment by gustavonihei** [comment]
@lhespress The CI is complaining that the efuse defconfig is not refreshed. Please, run `./tools/refresh.sh` to fix it.

File: arch/risc-v/src/esp32c3/esp32c3_efuse_lowerhalf.c [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+       * 'devpath' were not unique).  We know here but we return EEXIST to
+       * indicate the failure (implying the non-unique devpath).
+       */
+
+      ret = -EEXIST;
+      goto errout;
+    }
+
+errout:
+  return ret;
```
Comment (gustavonihei) [inline arch/risc-v/src/esp32c3/esp32c3_efuse_lowerhalf.c:242]:
```suggestion
    }

  return ret;
```
No need for this `goto`.

File: arch/risc-v/src/esp32c3/esp32c3_efuse_lowerhalf.c [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+ * Name: esp32c3_efuse_lowerhalf_ioctl;
+ *
+ * Description:
+ *   Initialize the efuse driver. The efuse is initialized
+ *   and registered as 'devpath'.
+ *
+ * Input Parameters:
+ *   lower        - A pointer the publicly visible representation of
+ *                  the "lower-half" driver state structure
+ *   cmd          - The ioctol command value
```
Comment (gustavonihei) [inline arch/risc-v/src/esp32c3/esp32c3_efuse_lowerhalf.c:167]:
```suggestion
 *   cmd          - The ioctl command value
```

File: arch/risc-v/src/esp32c3/esp32c3_efuse.h [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+    EFUSE_BLK_KEY_MAX          = 10,
+
+    EFUSE_BLK10                = 10,  /**< Number of eFuse BLOCK10. SYS_DATA_PART2 */
+    EFUSE_BLK_SYS_DATA_PART2   = 10,  /**< Number of eFuse BLOCK10. SYS_DATA_PART2 */
+
+    EFUSE_BLK_MAX
+} esp_efuse_block_t;
+
+/****************************************************************************
+ * Name: efuse_func_proc_t;
```
Comment (gustavonihei) [inline arch/risc-v/src/esp32c3/esp32c3_efuse.h:85]:
```suggestion
 * Name: efuse_func_proc_t
```

File: arch/risc-v/src/esp32c3/esp32c3_efuse.h [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+                                  int bit_start,
+                                  int bit_count,
+                                  void *arr, int *bits_counter);
+
+/****************************************************************************
+ * Public Functions Prototypes
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: esp32c3_efuse_read_field;
```
Comment (gustavonihei) [inline arch/risc-v/src/esp32c3/esp32c3_efuse.h:112]:
```suggestion
 * Name: esp32c3_efuse_read_field
```

File: arch/risc-v/src/esp32c3/esp32c3_efuse.h [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+ * Returned Value:
+ *   Zero (OK) is returned on success. Otherwise -1 (ERROR).
+ *
+ ****************************************************************************/
+
+int esp32c3_efuse_read_field(const efuse_desc_t *field[], void *dst,
+                             size_t dst_size_bits);
+
+/****************************************************************************
+ * Name: esp32c3_efuse_write_field;
```
Comment (gustavonihei) [inline arch/risc-v/src/esp32c3/esp32c3_efuse.h:131]:
```suggestion
 * Name: esp32c3_efuse_write_field
```

File: arch/risc-v/src/esp32c3/esp32c3_efuse.h [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+ * Returned Value:
+ *   Zero (OK) is returned on success. Otherwise -1 (ERROR).
+ *
+ ****************************************************************************/
+
+int esp32c3_efuse_write_field(const efuse_desc_t *field[],
+                              const void *src, size_t src_size_bits);
+
+/****************************************************************************
+ * Name: esp32c3_efuse_burn_efuses;
```
Comment (gustavonihei) [inline arch/risc-v/src/esp32c3/esp32c3_efuse.h:150]:
```suggestion
 * Name: esp32c3_efuse_burn_efuses
```

File: arch/risc-v/src/esp32c3/esp32c3_efuse.h [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+ *
+ * Returned Value:
+ *   None.
+ *
+ ****************************************************************************/
+
+void esp32c3_efuse_burn_efuses(void);
+
+/****************************************************************************
+ * Name: esp32c3_efuse_initialize;
```
Comment (gustavonihei) [inline arch/risc-v/src/esp32c3/esp32c3_efuse.h:166]:
```suggestion
 * Name: esp32c3_efuse_initialize
```

File: arch/risc-v/src/esp32c3/esp32c3_efuse_lowerhalf.c (hunk 2 above)
Comment (lhespress) [reply arch/risc-v/src/esp32c3/esp32c3_efuse_lowerhalf.c]: Thanks, I have modified it.

File: arch/risc-v/src/esp32c3/esp32c3_efuse_lowerhalf.c (hunk 1 above)
Comment (lhespress) [reply arch/risc-v/src/esp32c3/esp32c3_efuse_lowerhalf.c]: Thanks, I have modified it.

File: arch/risc-v/src/esp32c3/esp32c3_efuse.h (hunk 3 above)
Comment (lhespress) [reply arch/risc-v/src/esp32c3/esp32c3_efuse.h]: Thanks, I have modified it.

File: arch/risc-v/src/esp32c3/esp32c3_efuse.h (hunk 4 above)
Comment (lhespress) [reply arch/risc-v/src/esp32c3/esp32c3_efuse.h]: Thanks, I have modified it.

File: arch/risc-v/src/esp32c3/esp32c3_efuse.h (hunk 5 above)
Comment (lhespress) [reply arch/risc-v/src/esp32c3/esp32c3_efuse.h]: Thanks, I have modified it.

File: arch/risc-v/src/esp32c3/esp32c3_efuse.h (hunk 6 above)
Comment (lhespress) [reply arch/risc-v/src/esp32c3/esp32c3_efuse.h]: Thanks, I have modified it.

File: arch/risc-v/src/esp32c3/esp32c3_efuse.h (hunk 7 above)
Comment (lhespress) [reply arch/risc-v/src/esp32c3/esp32c3_efuse.h]: Thanks, I have modified it.

**Review by gustavonihei — APPROVED** [review]

**Pull request merged**